docs/proposals/policy-feg-api-cleanup/qos_enforcement.md

```diff
@@ -0,0 +1,343 @@
+
+GET     /feg_lte/{network_id}/qos_profiles
+POST    /feg_lte/{network_id}/qos_profiles
+
+GET     /feg_lte/{network_id}/qos_profiles/{qos_name}
+PUT     /feg_lte/{network_id}/qos_profiles/{qos_name}
+DELETE  /feg_lte/{network_id}/qos_profiles/{qos_name}
+
+
+
+APN Profiles
+
+GET     /lte/{network_id}/apns
+POST    /lte/{network_id}/apns
+
+GET     /lte/{network_id}/apns/{apn_name}
+PUT     /lte/{network_id}/apns/{apn_name}
+DELETE  /lte/{network_id}/apns/{apn_name}
+
+GET     /feg_lte/{network_id}/apns
+POST    /feg_lte/{network_id}/apns
+
+GET     /feg_lte/{network_id}/apns/{apn_name}
+PUT     /feg_lte/{network_id}/apns/{apn_name}
+DELETE  /feg_lte/{network_id}/apns/{apn_name}
+
+
+
+APN Resources
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels
+POST    /lte/{network_id}/gateways/{gateway_id}/resource_labels
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+PUT     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+DELETE  /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+
+GET     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels
+POST    /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels
+
+GET     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+PUT     /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+DELETE  /feg_lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+```
+
+
+
+### API Entity Definitions
+
+Entity definitions for QoS, APN, and subscribers will change. 
+The final state is summarized here:
+
+```
+policy_qos_profile:
+  id: string
+  class_id: enum[0,255]
+  max_req_bw_dl: uint32
+  max_req_bw_ul: uint32
+  # Guaranteed bit rate
+  # Not enforced by Magma yet
+  gbr:
+    gbr_ul: uint32
+    gbr_dl: uint32
+  # Allocation and retention priority
+  # Not enforced by Magma yet
+  arp:
+    arp_priority_level: uint32
+    arp_preemption_capability: bool
+    arp_preemption_vulnerability: bool
+```
+
+This is an addition, as there is an existing `qos_profile` definition, which
+will continue to be used only for the `apn_configuration` entity.
+
+```
+apn_configuration:
+  # No change here
+  qos_profile: qos_profile 
+  # Aggregated Maximum bit rate
+  # for non-GBR bearers, this limits the aggregate bit rate
+  # Enforce this on pipelined
+  ambr: aggregated_maximum_bitrate
+  # resource_label should match the name of a specified ApnResource
+  resource_label: string
+  # Unsupported as this time
+  # To be mandatory
+  pdn_type: enum[IPV4, IPV6, IPV4_IPV6]
+```
+
+As before, a single QoS profile will be associated to an APN object at creation.
+This will define the QoS for the default bearer, in combination with the `ambr`
+field.
+
+```
+apn_resource:
+  id: string
+  # Determines the IPs that will be allocated to connected subscribers under
+  # this APN resource
+  ipv4_block: string 
```

Comment:
We are not planning on supporting per APN IP block  in next release. It might be better to leave this configuration for later time.